src/main/java/io/kestra/plugin/azure/messaging/eventhubs/Produce.java

```diff
@@ -0,0 +1,205 @@
+                                          "tweet": row.tweet
+                                        }
+                                      };
+                                      row = result
+                                  - id: sendToEventHubs
+                                    type: io.kestra.plugin.azure.messaging.eventhubs.Produce
+                                    from: "{{ outputs.rowToJson.uri }}"
+                                    bodySerde: "JSON"
+                                    eventHubName: from-kestra
+                                    connectionString: "{{ secrets.azure.eventhubs.connectionString }}"
+                                    maxBatchSizeInBytes: 4096
+                                    maxEventsPerBatch: 100
+                                    bodyContentType: application/json
+                                    eventProperties:
+                                        source: kestra
+                                """
+                }
+        )
+})
+@Schema(
+        title = "Publish events to Azure Event Hubs."
+)
+@Slf4j
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+public class Produce extends AbstractEventHubProduceTask implements RunnableTask<Produce.Output> {
+
+    // TASK'S METRICS
+    private static final String METRIC_SENT_EVENTS_NAME = "total-sent-events";
+    private static final String METRIC_SENT_BATCHES_NAME = "total-sent-batches";
+
+    // SERVICES
+    private final EventHubClientFactory clientFactory;
+    private final EventHubClientConfig.Factory<EventHubClientInterface, EventHubClientConfig> clientConfigFactory;
+
+    /**
+     * Empty constructor. Required by Kestra.
+     */
+    Produce() {
+        this(new DefaultAzureClientFactory(), new DynamicEventHubClientConfig.Factory<>());
+    }
+
+    /**
+     * Creates a new {@link Produce} instance using the specified client factory.
+     *
+     * @param clientFactory The client factory.
+     */
+    Produce(final EventHubClientFactory clientFactory,
+            final EventHubClientConfig.Factory<EventHubClientInterface, EventHubClientConfig> clientConfigFactory) {
+        this.clientFactory = Objects.requireNonNull(clientFactory, "clientFactory should not be null");
+        this.clientConfigFactory = Objects.requireNonNull(clientConfigFactory, "clientConfigFactory should not be null");
+    }
+
+    /**
+     * {@inheritDoc}
+     **/
+    @Override
+    @SuppressWarnings("unchecked")
+    public Output run(RunContext runContext) throws Exception {
+        final InputDataReader reader = new InputDataReader(runContext);
+
+        InputStream is = null;
+        if (this.getFrom() instanceof String data) {
+            is = reader.read(data);
+        }
+
+        if (this.getFrom() instanceof Map data) {
+            is = reader.read(data);
+        }
+
+        if (this.getFrom() instanceof List data) {
+            is = reader.read(data);
+        }
```

Comment:
We usually produce one message by 'line' as we handle data in micro-batch.
If it's a List: one message by element.
If it's a Map, we considered it is an object so a single message.
If it's a String, we consider it is an ION file so one message by line. In this case you should use the FileSerde.read() to read lines and read it in a reactive manner to avoid mounting everything in memory in case the file is big.
You can have a look at an example here: https://github.com/kestra-io/plugin-aws/blob/master/src/main/java/io/kestra/plugin/aws/sns/Publish.java#L68

Reply:
Yes, basically the [InputDataReader](https://github.com/kestra-io/plugin-azure/blob/5844107e38ad167757176d24bdbdaea7a8b98381/src/main/java/io/kestra/plugin/azure/eventhubs/internal/InputDataReader.java) is only responsible for providing an `IntputStream` of ION file whatever the type of input. 
Then the [`EventHubProducerService`](https://github.com/kestra-io/plugin-azure/pull/56/files#diff-5090d4c42e7117e826975702607e330eca66193ded5f03990aa2bb723888a43cR57-R69) uses a reactive approach to read that file and produce one event per line.